pkg/datapath/linux/probes/probes.go

```diff
@@ -599,6 +599,19 @@ func HaveIPv6Support() error {
 	return nil
 }
 
+// HaveBatchLookupSupport checks if the kernel supports batch lookup for the passed map.
+// Note, key and value types for the map must be passed in the generic form.
+func HaveBatchLookupSupport[K, V any](m *ebpf.Map) bool {
```

Comment:
Currently this probe programs against the underlying feature probe that `BatchLookup` uses under the hood: https://github.com/cilium/ebpf/blob/7719d2ffbbb8b3a595ef747b016939c8c7419e56/syscalls.go#L212

`haveBatchAPI` determines the general availability of the batch APIs in the kernel. If that feature test is successful we also return success because we only fail the probe if the returned error is `ErrNotSupported`. If that is what we want from this feature probe we can probably just copy the code of `haveBatchAPI` into the probes package and add some caching there directly (or export `haveBatchAPI` in cilium/ebpf cc @lmb). That would simplify the function signature here ( e.g. `HaveBatchSupport() error`) and would eliminate the need for any additional caching implemented in the cilium code for the logic that depends on this probe.

Alternatively, we could extend this probe to make sure to check that the `BatchLookup` call returned the exact values/errors we expect from making the `BatchLookup` call on the passed Map. But I haven't given much thought to how that would look like generically.

Reply:
I thought that `haveBatchAPI` only checks for map type `Hash`, [here](https://github.com/cilium/ebpf/blob/7719d2ffbbb8b3a595ef747b016939c8c7419e56/syscalls.go#L215). 

haveBatchAPI support in kernel is not consistent across map types, this probe method helps in identifying it. 

When using strace, I see that syscall makes it to kernel and it returns ENOTSUPP for batch lookup of map `HashOfMap`.

```
bpf(BPF_MAP_LOOKUP_BATCH, {batch={in_batch=NULL, out_batch=0xc000ae7960, keys=0xc000ae7950, values=0xc000ae7954, count=1, map_fd=7, elem_flags=BPF_ANY, flags=0}}, 56) = -1 ENOTSUPP (Unknown error 524)
```

Reply:
> I would prefer this approach, e.g have a HaveBatchLookup(mt ebpf.MapType) error API. 

Agreed, I think it's nicer to have this functionality provided in cilium/ebpf and do this once at startup and cache the result. 

To unblock this PR, should I move this method into multicast maps pkg for now.  Once change is introduced in cilium/ebpf, multicast code can be modified to use `HaveBatchLookup`.

Reply:
yes, ideally it should also stay unexported. Then we can just use the new API in cilium/ebpf as a drop-in replacement. The API in cilium/ebpf will likely also come with caching built-in, like the other feature probe APIs.

Reply:
I removed probes changes from this PR. `haveBatchLookupSupport` is moved to multicast maps package for now. Will work on cilium/ebpf for adding `haveBatchLookupSupport` in more generic way.

cc @ldelossa  I think we can take this apporach for now, wdyt?